### DISCUSSION TITLE
read additional pins using binary sensor example

### DISCUSSION THREAD

**Discussion by rsettle** [dawidchyrzynski/arduino-home-assistant#73] [Q&A]
I can get the example to run and be recognized by Home Assistant, but haven't been able to work out a way to read additional pins. Is there an easy way to modify this to read additional pins?

**Comment by pvoncken** [comment]
Hi rsettle,

That is possible.
I have a project using 5 Analog 4 digital sensors 4 buttons 9 switches and 4 fans.
I atached the code.
The only problem i have is that the code gets instable when i use the TempCalc function (to convert analog input value to degrees Celsius)
I didn't get that to work yet. If i leave the funtion out it works perfectly. 


Pascal
[Arduino-code.txt](https://github.com/dawidchyrzynski/arduino-home-assistant/files/8411969/Arduino-code.txt)
